src/hip_intercept.cpp

```diff
@@ -0,0 +1,43 @@
+#include <hc_am.hpp>
+
+#include "hsa/hsa.h"
+#include "hsa/hsa_ext_amd.h"
+#include "hip/hip_runtime.h"
+#include "hip_hcc_internal.h"
+#include "trace_helper.h"
+#include "hip/hip_cbapi.h"
+
+///////////////////////////////////////////////////////////////////////////////////
+// HIP API callbacks/activity
+
+extern "C" size_t HSAOp_get_record_size();
```

Comment:
Having to `extern "C"` is not a good idea. Before long HIP runtime may need to adopt another interface called VDI. These interfaces breaks such abstraction and mandates HIP runtime be coupled with existing HCC runtime.